tests/withBlockTracker.ts

```diff
@@ -32,27 +35,27 @@ type WithSubscribeBlockTrackerCallback = (args: {
 
 /**
  * An object that allows specifying the behavior of a specific invocation of
- * `sendAsync`. The `methodName` always identifies the stub, but the behavior
- * may be specified multiple ways: `sendAsync` can either return a promise or
+ * `request`. The `methodName` always identifies the stub, but the behavior
+ * may be specified multiple ways: `request` can either return a promise or
  * throw an error, and if it returns a promise, that promise can either be
  * resolved with a response object or reject with an error.
  *
  * @property methodName - The RPC method to which this stub will be matched.
- * @property response - Instructs `sendAsync` to return a promise that resolves
+ * @property response - Instructs `request` to return a promise that resolves
  * with a response object.
  * @property response.result - Specifies a successful response, with this as the
  * `result`.
  * @property response.error - Specifies an error response, with this as the
  * `error`.
- * @property implementation - Allows overriding `sendAsync` entirely. Useful if
+ * @property implementation - Allows overriding `request` entirely. Useful if
  * you want it to throw an error.
- * @property error - Instructs `sendAsync` to return a promise that rejects with
+ * @property error - Instructs `request` to return a promise that rejects with
  * this error.
  */
 type FakeProviderStub =
   | {
       methodName: string;
-      response: { result: any } | { error: string };
+      response: any;
```

Comment:
The way I used the word "response" was to refer to a response object. We don't have that anymore, because `request` more or less hides the response object. We have either a value that we want `request` to return (which comes from the `result` of the response object), or we have an error that we want it to throw.

Maybe we should name this `result` instead of `response` so that if someone doesn't know exactly what this is doing they at least won't be misled into thinking they're overriding the response object?